src/Workspaces/CSharp/Portable/Extensions/SyntaxNodeExtensions.cs

```diff
@@ -1021,5 +1021,9 @@ public static bool IsInDeconstructionLeft(this SyntaxNode node, out SyntaxNode d
             deconstructionLeft = null;
             return false;
         }
+
+
+        public static TypeSyntax StripRefIfNeeded(this TypeSyntax type)
+            => type is RefTypeSyntax refType ? refType.Type : type;
```

Comment:
ideally would go in a file/type called TypeSyntaxExtensions.   But that's very minor and having this here is fine.